### ISSUE TITLE
Values can't be found when used as python map keys

### ISSUE THREAD

**Issue by yurivict** [revng/llvmcpy#17]
When I add values as keys to the map:
```
mm = {}
mm[llvmValue] = ...
```
They can't be found later with ```mm.has_key[llvmValue]```.
Using ```in_ptr()``` works though:
```
mm = {}
mm[llvmValue.in_ptr()] = ...
mm.has_key[llvmValue.in_ptr()]  # finds it
```

I think, the python object for value isn't defined properly. There should be no need to use ```in_ptr()```, and should be no difference between both cases.

**Comment by aleclearmind** [comment]
I cannot reproduce this.

```
In [18]: dictionary = {}
In [19]: dictionary[func] = 3
In [20]: dictionary
Out[20]: {<llvmcpyimpl.Value at 0x7f83c8052490>: 3}
In [21]: dictionary[func]
Out[21]: 3
In [23]: dictionary.has_key(func)
Out[23]: True
```

Can you provide a repeatable test case, Python version, and llvmcpy commit you're using?

**Comment by yurivict** [comment]
Here's the testcase:
1. Extract the attached shar: 
[testcase.shar.txt](https://github.com/revng/llvmcpy/files/1608608/testcase.shar.txt)
2. cd testcase
3. Compile testcase.c to testcase.bc using ```clang39 -c -emit-llvm testcase.c```
4. Run:
```
$ python2 testcase.py 
MAP N
MAP N
MAP Y
MAP Y
```
5. In testcase.py, in the function ll2key, uncomment the first line, and comment the second one, and rerun:
```
$ python2 testcase.py 
MAP N
MAP N
MAP N
MAP N
```

The second run uses pure python objects as map keys, and map lookups fail (```N```).
The first run substitutes pure python objects with the pointers, in_ptr(), as map keys, and map lookup succeed (```Y```).

**Comment by aleclearmind** [comment]
Mmh, in my case was probably working because they were exactly the same object, instead in your case there's probably some copy going on. I'll add an `__hash__` method that returns the `in_ptr`.

**Comment by aleclearmind** [comment]
Should be fixed in 2ccead096bbe22bd088d72bc31339329a3d9a405.
Please verify and feel free to reopen this issue in case of problems.
Thanks for reporting this.
